flickr_download/flick_download.py

```diff
@@ -100,6 +100,23 @@ def download_set(set_id, get_filename, size_label=None, skip_download=False):
     download_list(pset, pset.title, get_filename, size_label, skip_download)
 
 
+def valid_filesystem_name(orig_basename):
+    """
+    Derive a valid file/directory name from the 'orig_basename'.
+
+    Unix filesystems allow almost anything but '/' in the filename,
+    but Windows has many restrictions: https://stackoverflow.com/a/31976060/25450
+    """
+    import re
+    forbiddenchars = re.compile(u"([/\\\\<\\>:\"\\|\\?\\*]|[\x00-\x1F])", re.UNICODE)
```

Comment:
There is no need to `compile` this first, and then do the `sub` since you are only using the expression once. Just do it in one go. That makes it easier to read